packages/toolpad-app/src/server/auth.ts

```diff
@@ -154,14 +154,14 @@ export async function createAuthPagesMiddleware(project: ToolpadProject) {
 
     const hasAuthentication = authProviders.length > 0;
 
+    const isPageRequest = req.get('sec-fetch-dest') === 'document';
     const signInPath = `${base}/signin`;
 
-    let isRedirect = false;
+    let isAuthorized = true;
     if (
+      (!project.options.dev || isPageRequest) &&
```

Comment:
In dev mode it only redirects pages so that the user in preview sees the same behavior there will be in production. Can't hide RPC requests behind authentication in dev mode for example, as the editor needs to make them.